crowbar_framework/app/models/node_object.rb

```diff
@@ -256,6 +256,32 @@ def intended_role=(value)
     @node["crowbar_wall"]["intended_role"] = value
   end
 
+  def disk_roles=(value)
+    @node["crowbar_wall"] ||= {}
+    @node["crowbar_wall"]["disk_roles"] = value
+  end
+
+  def disk_roles
+    disk_roles = crowbar_wall["disk_roles"] || {}
+
+    # remove devices that were claimed after they role was set
+    disk_roles.each do |d, _role|
+      crowbar_wall["disk_roles"].delete d if crowbar_wall[:claimed_disks].key? d
```

Comment:
Would you mind adding `()`'s around the method arguments, please? Makes the stuff a bit easier to read.